### PR TITLE
Bump flake and fix wasm builds; CI improvements

### PR DESCRIPTION
**Pull request by steveej**
please see commit messages for details

### CODE REVIEW

**Comment by jost-s** [comment]
@steveej It looks fine to me but the tests are failing.

**Comment by steveej** [comment]
> @steveej It looks fine to me but the tests are failing.

yes it's not fine yet. the workaround clears the env of something that's apparently needed. i couldn't figure out yet what needs to be removed from the env and stop inheritance for the `Command`. that's be preferable

File: test/build.rs
Comment (steveej) [inline test/build.rs:14]:
```suggestion
```

File: test/build.rs
Comment (steveej) [inline test/build.rs:30]:
```suggestion
```

**Review by steveej — COMMENTED** [review]
cleanup

**Comment by steveej** [comment]
@jost-s i figured out the env var that needs to be cleared. if you approve i'll clean up the commit history before merging.

**Review by jost-s — APPROVED** [review]

**Comment by jost-s** [comment]
@steveej Wait, this seems like a problem with the `rustDev` shell. Instead of working around it here, should it not be removed there?

**Comment by jost-s** [comment]
I'll merge this as a workaround to unblock the release.

**Comment by steveej** [comment]
> @steveej Wait, this seems like a problem with the `rustDev` shell. Instead of working around it here, should it not be removed there?

the inner `cargo build --target wasm32-unknown-unknown` spawned from the build.rs runs in the context of the outer `cargo build [--target x86_64-linux-unknown-gnu]`. 

for that target, the [`rustDev` shell sets these flags: `export CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_RUSTFLAGS="$CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_RUSTFLAGS -Clink-arg=-fuse-ld=mold"`](https://github.com/holochain/holochain/blob/a7a7ff34b6477f87114fdb9ad29d50b90de1899c/nix/modules/devShells.nix#L184).

the effect of it is the environment variable that i removed from the inner job in this PR, because we don't intend that effect for the wasm target.

let me know if that explains it for you.

**Comment by jost-s** [comment]
@steveej In effect does that mean that mold cannot link wasm targets? And why is that outer env var's target x86_64 even applied for wasm32?

**Comment by steveej** [comment]
> In effect does that mean that mold cannot link wasm targets?

true.



> And why is that outer env var's target x86_64 even applied for wasm32?

because of this, specifically the bold part

> the inner `cargo build --target wasm32-unknown-unknown` spawned from **the build.rs runs in the context of the outer `cargo build [--target x86_64-linux-unknown-gnu]`** (i augmented the target architecture spec which is implied by the ubuntu-latest CI runner).

**Comment by jost-s** [comment]
@steveej  But the target platforms are different, it's x86_64 in one and wasm32 in the other. Why a specific target triple if it's applied in a general way? Is the x86_64 applied anyway despite the difference in platform?

**Comment by steveej** [comment]
1. `nix develop` spawns a shell that will have `CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_RUSTFLAGS -Clink-arg=-fuse-ld=mold"` set
2. `cargo build [--target x86_64-linux-unknown-gnu]` discovers `CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_RUSTFLAGS=...` in its environment  and translates this to `CARGO_ENCODED_RUSTFLAGS=...` for itself and its spawned children. 
3. while processing the `test/build.rs`, `cargo build --target wasm32-unknown-unknown` is spawned, which discovers `CARGO_ENCODED_RUSTFLAGS=...` in its environment and thus uses `mold` accordingly.

does that help?

**Pull request merged**
